genomepy/annotation.py

```diff
@@ -714,6 +709,60 @@ def map_locations(self, annot: Union[str, pd.DataFrame], to: str) -> pd.DataFram
         return df
 
 
+@memory.cache
```

Comment:
how is joblib different from the BucketCache we already have for genomepy? [example init](https://github.com/vanheeringen-lab/genomepy/blob/76f9c3b8b849332ef6f9f767fdb98c78e5f36b46/genomepy/provider.py#L55), [example usage](https://github.com/vanheeringen-lab/genomepy/blob/76f9c3b8b849332ef6f9f767fdb98c78e5f36b46/genomepy/provider.py#L666).

Reply:
BucketCache didn't recognize that it had already seen a `list` or `pd.Series`. This worked flawlessy.

Reply:
Hmm, but it would a bit too much to use two different caching libraries. However, `joblib.Memory` doesn't have an `expiry`-like option, it caches "forever". BucketCache on the other hand doesn't recognize the input arguments as the same, so it essentially doesn't cache.

Reply:
Neither caching method is optimal. I'm looking for one that 
- is timed
- can store objects as well as methods/functions
- is thread safe (optional)
- fast at retrieving (loading NCBI genomes is super slow right now)

Reply:
First possible solution I found:
Maybe this function (and the prover genomes) could be wrapped with `@lru_cache`? If I got it correctly it only lasts the session, but is in memory (so faster that pickling with bucketcache).

The ouput of this function is a pandas dataframe, so the user can just use `.to_csv()` for long term storage.

Reply:
For this use-case I would actually like a more permanent cache. This means that if you map gene symbols over multiple runs of a tool, it will just re-use the cache.

Reply:
Actually, `joblib.Memory` performs fine, it's just the downside of having two caching lib dependencies.

Reply:
I did a quick check, 'joblib.Memory` does *not* do well on NCBI genomes caching.

Reply:
Second option: memcached can handle timeouts, but we'd have to write some long term (>session, <several days) storage wrappers.

Third option: joblib seems to have timed in-memory caching, and pickling commands built-in, just no timer. We can make a wrapper that checks for pickles first, recreates old ones, then stores the objects in memory.

Reply:
Fourth option: [dogpile cache](https://dogpilecache.sqlalchemy.org/en/latest/usage.html#rudimentary-usage). This was suggested by the maker of bucketcache when i ran into threading issues.

Reply:
It will all mean major testing and rewrites though. The funny thing is that bucketcache claims they're thread-safe. 

For now, why not be pragmatic, and use two libraries. Let's safe testing different caching libs for later.

Reply:
agreed

Reply:
> 
> 
> The funny thing is that bucketcache claims they're thread-safe.

maybe I'm confusing terminology. Errors can occur when multiple uses of bucketcache run in parallel, and the cache expires/is deleted in one of the runs. [We discussed it in this issue](https://github.com/RazerM/bucketcache/issues/10).

Reply:
Sorry, I was confuse with [diskcache](https://pypi.org/project/diskcache/).